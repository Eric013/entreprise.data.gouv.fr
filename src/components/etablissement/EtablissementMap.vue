<template>
  <div v-if="mapboxglSupported" class="panel" id="map" ref="map"></div>
  <div v-else class="panel">
    <p class="panel__message">Votre navigateur ne supporte pas WebGL et ne peut pas afficher la carte de l'Ã©tablissement.</p>
  </div>
</template>

<script>
import Vue from 'vue'
import mapboxgl from 'mapbox-gl'
import mapOtherMarkers from '@/components/mixins/mapOtherMarkers'

export default {
  name: 'EtablisssementMap',
  props: ['positionEtablissement', 'siret'],
  data () {
    return {
      mapboxglSupported: mapboxgl.supported(),
      mapTilesEtalab: 'https://openmaptiles.geo.data.gouv.fr/styles/osm-bright/style.json',
      mapOptions (json) { 
        return {
          container: 'map',
          style: json.body,
          center: this.positionEtablissement,
          zoom: 13
        }
      },
      markers:  {
        red: { color: '#C0392B' },
        darkBlue: { color: '#274492' },
        blue: { color: '#0081d5' },
        lightBlue: { color: '#1e90da' }
      }
    }
  },
  mounted () {
    Vue.http.get(this.mapTilesEtalab).then((json) => { this.initMap(json) })
  },
  methods: {
    initMap: async function (json) {
      let map = await new mapboxgl.Map(this.mapOptions(json))
      // addOtherMarkets first so the etablissement marker will be on top
      this.addOtherMarkers(map, this.siret)
      this.addEtablissementMarker(map)
    },
    addEtablissementMarker (map) {
      new mapboxgl.Marker(this.markers.red)
        .setLngLat(this.positionEtablissement) 
        .addTo(map)
    }
  },
  mixins: [mapOtherMarkers]
}
</script>

<style lang="scss" scoped>
  #map {
    padding: 0;
    height: 400px;
  }

  .panel {
    width: 100%
  }

  .panel__message {
    color: $color-dark-grey
  }
</style>
