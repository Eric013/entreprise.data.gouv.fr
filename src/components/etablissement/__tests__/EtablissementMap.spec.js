import EtablissementMap from '@/components/etablissement/EtablissementMap.vue'
import { createLocalVue, mount } from '@vue/test-utils'
import Vue from 'vue';
import VueResource from 'vue-resource';

Vue.use(VueResource);

jest.mock('mapbox-gl', () => jest.fn())

describe('EtablissementMap.vue snapshot', () => {

  const localVue = createLocalVue()

  const wrapperEtablissementMap = mount(EtablissementMap, {
    localVue
  })

  test('It match the snapshot', () => {
    expect(wrapperEtablissementMap.vm.$el).toMatchSnapshot()
  })
})