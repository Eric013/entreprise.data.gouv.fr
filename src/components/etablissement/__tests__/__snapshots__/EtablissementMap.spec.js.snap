// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EtablissementMap.vue snapshot It match the snapshot 1`] = `
<div
  class="panel"
  id="map"
/>
`;
